packages/core/src/utils/SIWXUtil.ts

```diff
@@ -1,4 +1,256 @@
 import type { CaipNetworkId } from '@reown/appkit-common'
+import { OptionsController } from '../controllers/OptionsController.js'
+import { CoreHelperUtil } from './CoreHelperUtil.js'
+import { ChainController } from '../controllers/ChainController.js'
+import { ConnectionController } from '../controllers/ConnectionController.js'
+import { ModalController } from '../controllers/ModalController.js'
+import { SnackController } from '../controllers/SnackController.js'
+import { RouterController } from '../controllers/RouterController.js'
+import UniversalProvider from '@walletconnect/universal-provider'
+import { EventsController } from '../controllers/EventsController.js'
+import { AccountController } from '../controllers/AccountController.js'
+import { W3mFrameRpcConstants } from '@reown/appkit-wallet'
+import { StorageUtil } from './StorageUtil.js'
+
+export const SIWXUtil = {
+  getSIWX() {
+    return OptionsController.state.siwx
+  },
+  async initializeIfEnabled() {
+    const siwx = OptionsController.state.siwx
+    const address = CoreHelperUtil.getPlainAddress(ChainController.getActiveCaipAddress())
+    const network = ChainController.getActiveCaipNetwork()
+
+    if (!(siwx && address && network)) {
+      return
+    }
+
+    try {
+      const sessions = await siwx.getSessions(network.caipNetworkId, address)
+      if (sessions.length) {
+        return
+      }
+
+      await ModalController.open({
+        view: 'SIWXSignMessage'
+      })
+    } catch (error: unknown) {
+      // eslint-disable-next-line no-console
+      console.error('SIWXUtil:initializeIfEnabled', error)
+
+      EventsController.sendEvent({
+        type: 'track',
+        event: 'SIWX_AUTH_ERROR',
+        properties: this.getSIWXEventProperties()
+      })
+
+      // eslint-disable-next-line no-console
+      await ConnectionController._getClient()?.disconnect().catch(console.error)
+      RouterController.reset('Connect')
+      SnackController.showError('A problem occurred while trying initialize authentication')
+    }
+  },
+  async requestSignMessage() {
+    const siwx = OptionsController.state.siwx
+    const address = CoreHelperUtil.getPlainAddress(ChainController.getActiveCaipAddress())
+    const network = ChainController.getActiveCaipNetwork()
+    const client = ConnectionController._getClient()
+
+    if (!(siwx && address && network && client)) {
+      throw new Error('SIWX is not enabled')
+    }
```

Comment:
why throw here but not on the initialization?

Reply:
because this function should not be called if siwx was not enabled/initialized